internal/nexodus/stun.go

```diff
@@ -0,0 +1,36 @@
+package nexodus
+
+import (
+	_ "embed"
+	"math/rand"
+	"strings"
+	"sync"
+)
+
+//go:embed stun-servers.txt
+var stunServersTxtFile string
+var currentStunServer = 0
+var stunServerMu = sync.Mutex{}
+
+var (
+	stunServers = []string{}
+)
+
+func init() {
+	stunServers = strings.Split(stunServersTxtFile, "\n")
+	for i := range stunServers {
+		stunServers[i] = strings.TrimSpace(stunServers[i])
+	}
+	// #nosec G404
+	currentStunServer = rand.Intn(len(stunServers))
```

Comment:
A minor improvement would be to use `rand.Shuffle()` so that every `nexd` instance isn't going through the list in the same order.

https://pkg.go.dev/math/rand#Shuffle